release-notes/release-notes-v0.5.0.md

```diff
@@ -0,0 +1,805 @@
+This does not yet allow you to load and use multiple wallets at once, but you can switch between wallets using the
+`bitcoin-s.wallet.walletName` config option.
+
+## Module Level Changes
+
+### Bitcoind RPC Client
+
+The Bitcoind RPC Client now has support for v0.20 & v0.21!
+
+The Bitcoind RPC Client now has support for using it with a bitcoind that has multiple wallets loaded, this is done by
+setting the `walletName` parameter in the required functions.
+
+#### Commits
+
+648e7d9abac Bitcoind v0.21.0 support (#2414)
+
+3f18ba18d7e Bitcoind Version from String (#2421)
+
+905491fe0e8 Fix getPeerInfo for v0.20 (#2402)
+
+e8305477739 Use PSBT type in bitcoind calls (#2242)
+
+aaf1a81d4ca Bitcoind RPC Multi-wallet support  (#2231)
+
+7c887cc144b Move newer functions to common rpc interface (#2230)
+
+a5621f5f565 Add bitcoind functions for load and unload wallet (#2229)
+
+ead4eaa1477 Make bitcoind extend chain api (#2087)
+
+206c5a93d00 Bitcoind v0.20 updated rpcs & tests (#2061)
+
+0b100383f32 Create initial BitcoindV20RpcClient (#2060)
+
+### Chain
+
+The `ChainHandler` has been split up into two different types: `ChainHandler` and `CachedChainHandler`.
+
+`CachedChainHandler` stores in memory the latest block headers and allows for faster syncing.
+`ChainHandler` uses the chain database to access all data, this is necessary for things like the wallet to be able to
+point to a single `ChainApi` and not get outdated data.
+
+#### Commits
+
+1dacb74edf7 Optimize filter sync and fetching filter heights (#2568)
+
+e44a620ea1a Small cleanups on the chain docs (#2515)
+
+4e285e67469 Make ChainApi.processHeaders() return a failed future in the case we … (#2436)
+
+9bef444fe70 Rework BlockHeaderDAO.chainTips into two methods: BlockHeaderDAO.{get… (#2443)
+
+40f46d7c7a8 Remove BlockHeader.getBlockchainsFrom(), rework type signature for Bl… (#2431)
+
+2de17bb4e47 2020 11 13 issue 2258 (#2260)
+
+### CLI
+
+New CLI commands have been added such as `decodepsbt`, `getblockheader`, and `bumpfeecpfp`. For a full list of available CLI
+commands check [the docs](https://bitcoin-s.org/docs/0.5.0/applications/server#server-endpoints).
+
+The `Cli` module is now published as a library, this should allow other projects to easily interact with a Bitcoin-S server.
+
+The `cli` is also published using the `sbt-native-image` plugin. (#2494)
+
+#### Commits
+
+f849ba492f1 Wallet name in walletinfo (#2603)
+
+94c71543fb9 Add createmultisig cli command (#2495)
+
+fb43748d978 Version Number in logs & Cli Command (#2467)
+
+2be2df12dbb Publish Cli as library (#2433)
+
+847981317a0 Add getblockheader cli command (#2424)
+
+f135322c094 Bump Fee Cli commands (#2415)
+
+a2b54eef308 Import Seed cli commands (#2376)
+
+ce9fbb0807a Analyze PSBT function (#2240)
+
+c08379b2361 Decode PSBT function (#2237)
+
+c662cf802d1 Make tx an argument for decoderawtransaction (#2155)
+
+8c4308e2c6a Make cli pretty print json (#2130)
+
+### Commons
+
+#### Commits
+
+### Core
```

Comment:
Also needs experimental support for DLC-related data structures and algorithms with a similar message to TLV support below